### PR TITLE
Fix header date format

### PR DESCRIPTION
**Pull request by mclrc**
The date format we use for the `If-Modified-Since` header is non-standard. This leads to issues on android because the network stack can add it's own caching headers with the specced format, which will result in a code 500 with current parsing logic.

See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since

### CODE REVIEW

File: packages/backend/utils_test/CheckIfModifiedSince_test.go [hunk 1]
```diff
@@ -27,14 +26,14 @@ func TestCheckIfModifiedSince(t *testing.T) {
 		},
 		{
 			name:             "Correct header, not modified",
-			ifModifiedSince:  "2024-01-01T12:00:00Z",
+			ifModifiedSince:  "Mon, 01 Jan 2024 12:00:00 GMT",
```
Comment (johan-t) [inline packages/backend/utils_test/CheckIfModifiedSince_test.go:29]: Should we be using GMT here or UTC?

**Review by johan-t — APPROVED** [review]
Good catch. Before merging please make sure that the caching still works, just in case as it is not being tested automatically.

File: packages/backend/utils_test/CheckIfModifiedSince_test.go (hunk 1 above)
Comment (mclrc) [reply packages/backend/utils_test/CheckIfModifiedSince_test.go:29]: very confusing but GMT at the end usually means UTC nowadays. check the output of `new Date().toUTCString()`.

Comment (johan-t) [reply packages/backend/utils_test/CheckIfModifiedSince_test.go:29]: > very confusing but GMT at the end usually means UTC nowadays. check the output of `new Date().toUTCString()`.

lol

**Pull request merged**